app/controllers/v1/submissions_controller.rb

```diff
@@ -12,6 +12,11 @@ def show
   def create
     task = Task.find(params.dig(:submission, :task_id))
 
+    if (submission = task.active_submission)&.completed? && !correcting_submission?
```

Comment:
Could you swap this round, so if `!correcting_submission` is 'true', it doesn't need to get the `active_submission`?